tests/integration/test_task_remote_mgr.py

```diff
@@ -0,0 +1,104 @@
+          this is picked out.
+        - Install targets where we can get a platform
+          - foo - Install target is implicitly the platfrom name.
+          - bar9 - The install target is implicitly the plaform name,
+            and the platform name matches a platform regex.
+          - baz - Install target is set explicitly.
+        - An install target (qux) where we cannot get a platform: Ensure
+          that we get the desired error.
+    """
+    mock_glbl_cfg(
+        'cylc.flow.platforms.glbl_cfg',
+        '''
+            [platforms]
+                [[foo]]
+                    # install target = foo  (implicit)
+                    # hosts = foo  (implicit)
+                [[bar.]]
+                    # install target = bar1 to bar9 (implicit)
+                    # hosts = bar1 to bar9 (implicit)
+                [[baz]]
+                    install target = baz
+                    hosts = baz
+        ''',
+    )
+
+    # Get a scheduler:
+    id_ = flow(one_conf)
+    schd = scheduler(id_)
+    async with start(schd) as log:
+        # Write database with 6 tasks using 3 platforms:
+        platforms = ['baz', 'bar9', 'foo']
+        line = r"('', '', {}, 0, 1, '', '', 0,'', '', '', 0, '', '{}', 4, '')"
+        stmt = r"INSERT INTO task_jobs VALUES" + r','.join([
+            line.format(i, platform) for i, platform in enumerate(platforms)
+        ])
+        schd.workflow_db_mgr.pri_dao.connect().execute(stmt)
+        schd.workflow_db_mgr.pri_dao.connect().commit()
+
+        # Mock a remote init map.
+        schd.task_job_mgr.task_remote_mgr.remote_init_map = {
+            'baz': REMOTE_FILE_INSTALL_DONE,     # Should match platform baz
+            'bar9': REMOTE_FILE_INSTALL_DONE,    # Should match platform bar.
+            'foo': REMOTE_FILE_INSTALL_DONE,     # Should match plaform foo
+            'qux': REMOTE_FILE_INSTALL_DONE,     # Should not match a plaform
+            'quiz': REMOTE_FILE_INSTALL_FAILED,  # Should not be considered
+        }
+
+        # Clear the log, run the test:
+        log.clear()
+        schd.task_job_mgr.task_remote_mgr.remote_tidy()
+        records = [str(r.msg) for r in log.records]
+
+        # We can't get qux, no defined platform has a matching install target:
+        qux_msg = (
+            'Unable to tidy the following install targets because no'
+            ' matching platform was found:\n * qux')
+        assert qux_msg in records
+
+        # We can get foo bar baz, and we try to remote tidy them.
+        # (This will ultimately fail, but past the point we are testing).
+        for target in ['foo', 'bar9', 'baz']:
+            msg = f'platform: {target} - remote tidy (on {target})'
+            assert msg in records
+
+        # We haven't done anything with Quiz because we're only looking
+        # at cases where platform == REMOTE_FILE_INSTALL_DONE
+        assert not [r for r in records if 'quiz' in r]
```

Comment:
Can dedent this bit (don't need the scheduler to be running any more:

```suggestion
    records = [str(r.msg) for r in log.records]

    # We can't get qux, no defined platform has a matching install target:
    qux_msg = (
        'Unable to tidy the following install targets because no'
        ' matching platform was found:\n * qux')
    assert qux_msg in records

    # We can get foo bar baz, and we try to remote tidy them.
    # (This will ultimately fail, but past the point we are testing).
    for target in ['foo', 'bar9', 'baz']:
        msg = f'platform: {target} - remote tidy (on {target})'
        assert msg in records

    # We haven't done anything with Quiz because we're only looking
    # at cases where platform == REMOTE_FILE_INSTALL_DONE
    assert not [r for r in records if 'quiz' in r]
```